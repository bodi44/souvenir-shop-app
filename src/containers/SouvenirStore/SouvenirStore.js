import React, { useEffect } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { fetchCurrencies } from '../../actions/currencies'
import {
  getCurrencies,
  isCurrenciesHasErrorLoading,
  isCurrenciesLoading
} from '../../reducers'

import App from '../../components/App'
import FilterRequest from '../FilterRequest/FilterRequest'
import VisibleOutput from '../VisibleOutput/VisibleOutput'

const SouvenirStore = ({ currencies, fetchCurrencies, isLoading, hasError }) => {
  useEffect(() => {
    fetchCurrencies()
  }, [fetchCurrencies])

  if (isLoading) return <App message={'Data is Loading...'}/>

  if (hasError) return <App message={'Error occurred while fetching API...'}/>

  return (
    <App>
      <FilterRequest currencies={currencies}/>
      <VisibleOutput/>
    </App>
  )
}

SouvenirStore.propTypes = {
  currencies: PropTypes.object.isRequired,
  fetchCurrencies: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
  hasError: PropTypes.any
}

export default connect(
  state => ({
    currencies: getCurrencies(state),
    isLoading: isCurrenciesLoading(state),
    hasError: isCurrenciesHasErrorLoading(state)
  }),
  {
    fetchCurrencies
  }
)(SouvenirStore)