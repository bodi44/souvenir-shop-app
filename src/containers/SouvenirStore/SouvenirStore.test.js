import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import expect from 'expect'
import Adapter from 'enzyme-adapter-react-16'

import SouvenirStore from './SouvenirStore'
import ProviderWrapper from '../../components/Provider'
import configureStore from '../../configureStore'

Enzyme.configure({ adapter: new Adapter() })

describe('SouvenirStore component', () => {
  it('should render with no props only wrapped in Provider', () => {
    const component = shallow(
      <ProviderWrapper store={configureStore()}>
        <SouvenirStore/>
      </ProviderWrapper>
    )

    expect(component).toMatchSnapshot()
  })
})