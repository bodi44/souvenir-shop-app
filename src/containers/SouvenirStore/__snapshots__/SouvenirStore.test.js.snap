// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SouvenirStore component should render with no props only wrapped in Provider 1`] = `ShallowWrapper {}`;
